### PR TITLE
Datasource documentation tune-up

### PR DESCRIPTION
**Pull request by kjdelisle**
connected to https://github.com/strongloop-internal/scrum-apex/issues/175

### CODE REVIEW

**Comment by kjdelisle** [comment]
@raymondfeng When you're feeling better, please take a look at the comments marked `XXX(kev)`; Most of them are questions for your clarification.

Once this portion is complete, I'll move onto tuning up the corresponding juggler documentation in [loopback.io](https://github.com/strongloop/loopback.io)

File: lib/datasource.js [hunk 1]
```diff
@@ -264,8 +264,12 @@ DataSource._resolveConnector = function(name, loader) {
 };
 
 /**
- * Connect to the data source
+ * Connect to the data source.
+ * If no callback is provided, it will return a Promise.
+ * Emits the 'connect' event.
  * @param callback
+ * @returns {Promise}
+ * @emits connect
```
Comment (raymondfeng) [inline lib/datasource.js]: The event name is `connected`.

File: lib/datasource.js [hunk 2]
```diff
@@ -451,10 +460,13 @@ DataSource.prototype.defineScopes = function(modelClass, scopes) {
   }
 };
 
-/*!
- * Define relations for the model class from the relations object
- * @param modelClass
- * @param relations
+/**
```
Comment (raymondfeng) [inline lib/datasource.js]: IIRC, we intentionally use `/*!` to not expose this method to generated API references.

File: lib/datasource.js [hunk 3]
```diff
@@ -731,23 +745,35 @@ DataSource.prototype.mixin = function(ModelCtor) {
 };
 
 /**
- * See ModelBuilder.getModel
+ *
+ * // XXX(kev): Deprecation notice for 4.x?
```
Comment (raymondfeng) [inline lib/datasource.js]: Do we want to use a generic tag, such as `// REVIEW` or `// FIXME`?

File: lib/datasource.js [hunk 4]
```diff
@@ -949,7 +990,7 @@ DataSource.prototype.autoupdate = function(models, cb) {
  * @property {Boolean} views If true, include views; if false, only tables.
  * @property {Number} limit Page size
  * @property {Number} offset Starting index
- *
+ * @returns {Model[]}
```
Comment (raymondfeng) [inline lib/datasource.js]: ModelDefinition[]

File: lib/datasource.js [hunk 5]
```diff
@@ -970,13 +1011,14 @@ DataSource.prototype.discoverModelDefinitions = function(options, cb) {
...
 /**
+ * FIXME Deprecate/remove this for LB 4.0?
  * The synchronous version of discoverModelDefinitions.
  * @options {Object} options The options
  * @property {Boolean} all If true, discover all models; if false, discover only models owned by the current user.
  * @property {Boolean} views If true, nclude views; if false, only tables.
  * @property {Number} limit Page size
  * @property {Number} offset Starting index
- * @returns {*}
+ * @returns {Model[]}
```
Comment (raymondfeng) [inline lib/datasource.js]: ModelDefinition[]

**Comment by kjdelisle** [comment]
@loay Don't forget to squash and rename the commits to remove "wip"

**Pull request merged**